### PR TITLE
Convert SQLite int8 column to ClickHouse int64

### LINKED ISSUE #40639: SQLite int8 column type conversion
**Describe what's wrong**

SQLite's `int8` column (8 bytes) gets read as `int32` column in ClickHouse. Large integers overflow as a result.

**Does it reproduce on recent release?**

```
clickhouse server --version
ClickHouse server version 22.9.1.404 (official build).
```

**How to reproduce**

To reproduce:
```sh
# create a table with sqlite
sqlite3 test.sqlite
sqlite> CREATE TABLE test (int8col int8);
sqlite> INSERT INTO test VALUES (2147483648);
sqlite> SELECT * FROM test;
2147483648

# query test.sqlite from ClickHouse
SELECT * FROM sqlite('test.sqlite', 'test');
> int8col
> -2147483648
```

**Expected behavior**

The column should be inferred as int64 column and read the original value without overflow.

**Additional context**

The issue seems to be stemming from this line where any column starting with `int` is converted to `int32`.
https://github.com/ClickHouse/ClickHouse/blob/master/src/Databases/SQLite/fetchSQLiteTableStructure.cpp#L38

### PR DESCRIPTION
**Pull request by barumrho**
SQLite int8 column uses 64 bit integers.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fixes SQLite int8 column conversion to int64 column in ClickHouse. Fixes #40639

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Ok. Need to add a test.

**Comment by barumrho** [comment]
> Ok. Need to add a test.

Looks like there was a test case that should have covered this, but had a bug due to column names being omitted.

File: tests/queries/0_stateless/01889_sqlite_read_write.reference [hunk 1]
```diff
@@ -21,7 +21,7 @@ line3	3
 2	text2
 3	text3
 test types
-CREATE TABLE SQLite.table4\n(\n    `a` Nullable(Int32),\n    `b` Nullable(Int32),\n    `c` Nullable(Int8),\n    `d` Nullable(Int16),\n    `e` Nullable(Int32),\n    `bigint` Nullable(String),\n    `int2` Nullable(String),\n    `int8` Nullable(String)\n)\nENGINE = SQLite
+CREATE TABLE SQLite.table4\n(\n    `a` Nullable(Int32),\n    `b` Nullable(Int32),\n    `c` Nullable(Int8),\n    `d` Nullable(Int16),\n    `e` Nullable(Int32),\n    `f` Nullable(Int64),\n    `g` Nullable(Int32),\n    `h` Nullable(Int64)\n)\nENGINE = SQLite
```
Comment (kssenii) [inline tests/queries/0_stateless/01889_sqlite_read_write.reference:24]: But it tests nothing as nothing is inserted / selected from tis table + fetchSQLiteTableStructure is used for table function, not for table engine.

Comment (kssenii) [reply tests/queries/0_stateless/01889_sqlite_read_write.reference:24]: ah, no, it is ok, sorry

**Pull request merged**